### ISSUE TITLE
How to regenerate metadata

### ISSUE THREAD

**Issue by nilolalolal** [HashLips/hashlips_art_engine#1219]
Hey there! 
I generated a collection, and now I want to change metadata only ( but I changed layer names). When I go utils/generate_metadata.js it gives me Error: Cannot find module 'C:\Users\komp\Desktop\hashlips_art_engine-main\utils\generate_metadata.js. 
can you help me with this? Thank you!

**Comment by bolshoytoster** [comment]
@nilolalolal the error is saying it can't find `utils\generate_metadata.js`, you're probably in the wrong folder (run `Get-ChildItem -Directory` to get the name, then run `cd ` + the name to change into it.

Besides, you probably want `utils/update_info.js` for the majority of uses.

What are you trying to do?
`generate_metadata.js` just replaces your metadata with preset values and `update_info.js` just updates `namePrefix`, `description` and `baseUri`.

**Comment by nilolalolal** [comment]
@bolshoytoster I generated a collection with wrong trait names, and now I changed names, and just want to generate metadata that it is compatible to already generated NFTs. 
But, I deleted all previous process in Studio Code because of other work I had to do. 
And now I'm not sure how to load generated NFTs and generate updated metadata for them. 
So basically, I'm not familiar with any step of how to do it..
Thank you @bolshoytoster

**Comment by bolshoytoster** [comment]
@nilolalolal
> now I changed names, and just want to generate metadata that it is compatible to already generated NFTs.

You won't be able to do this, as there's no way for it to know what traits to change to what.

You could try a find+replace in `_metadata.json` (you can either do this in most text editors with `ctrl` + `r`, or by running `(Get-Content build/json/_metadata.json) -Replace 'foo', 'bar' | Set-Content build/json/_metadata.json` where `foo` is what you want to replace and `bar` is what you want to replace it with (run this for each replacement).)

It's probably easiest to just regenerate the collection.

**Comment by nilolalolal** [comment]
@bolshoytoster oh okay, thank you! 
I'll generate it again.. 
Is there any chance I can speed up that? It takes 5 sec to generate 1 piece, and art is very simple. 
and for some more complicated takes 2 sec per piece..not sure why ?

**Comment by bolshoytoster** [comment]
@nilolalolal [my fork](https://github.com/bolshoytoster/hashlips_art_engine) has some speed optimisations.

**Comment by nilolalolal** [comment]
@bolshoytoster  thank you!

**Comment by TTibbs** [comment]
Hi I got a issue where in the metadata "ipfs://" was left out of "image": so it just has "image": "CIDHERE/1.png" for example. Any tips on fixing this?

**Comment by bolshoytoster** [comment]
@TTibbs you need to include `ipfs://` in your `baseUri`.

**Comment by TTibbs** [comment]
Fixed it the other day, just saw this now and thank you as well

**Comment by sameernoorani1** [comment]
Related question: I generated 1200 images and want to manually select the "best" 1000 (and discarding 200). After that's done I want to regenerate the metadata file for the remaining 1000 images. Anyway to do that?

**Comment by bolshoytoster** [comment]
@sameernoorani1 delete `build/json/_metadata.json`, then delete the individual json files for the editions you don't want.

Then run [my `utils/update_info.js`](https://raw.githubusercontent.com/bolshoytoster/hashlips_art_engine/main/utils/update_info.js) to remake `build/json/_metadata.json`.

**Comment by advra** [comment]
@bolshoytoster I deleted `_metadata.json` then ran `node ./utils/update_info.js` as you suggested but I get the error below. Are you certain we were supposed to delete the metadata?

```
Error: ENOENT: no such file or directory, open '/mnt/b/Downloads/Extra Files/Art/5-ether-frens/katsu/art-engine/build/json/_metadata.json'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at Object.<anonymous> (/mnt/b/Downloads/Extra Files/Art/5-ether-frens/katsu/art-engine/utils/updateInfo.js:27:18)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
```

**Comment by bolshoytoster** [comment]
@advra yes, you need to replace your `utils/update_info.js` with [mine](https://raw.githubusercontent.com/bolshoytoster/hashlips_art_engine/main/utils/update_info.js)

**Comment by advra** [comment]
I apologize. I misread. I was looking for a script which re-generated the images using existing json metadata not regenerate metadata itself.

**Comment by bolshoytoster** [comment]
@advra there isn't a script to do that because the image filenames aren't stored in the metadata.

**Comment by advra** [comment]
@bolshoytoster Strange. This makes back testing almost impossible. Truly the filenames could be stored in metadata for development purposes like regression testing when artist changes graphics, The additional data would then be removed using another script to to finalize the metadata.
